koku/reporting/migrations/0290_awsenabledcategorykeys_uuid.py

```diff
@@ -0,0 +1,49 @@
+# Generated by Django 3.2.19 on 2023-06-20 19:04
```

Comment:
The steps in this migration are far more complex than I was expecting and I am not sure we should ever try to run this much complexity in a single migration. Creating a new table, moving everything to it, dropping the old table, and then renaming the new table to the old table seems like way too many steps in one migration. Also, why are all of those steps necessary? 

If you set the model to this:
```
class AWSEnabledCategoryKeys(models.Model):
    """A collection of the current enabled category keys."""

    class Meta:
        """Meta for AWSCategoryTagKeys."""

        indexes = [models.Index(fields=["key", "enabled"], name="aws_enabled_category_key_index")]
        db_table = "reporting_awsenabledcategorykeys"

    uuid = models.UUIDField(default=uuid4, editable=False, unique=True, primary_key=True)
    key = models.CharField(max_length=253)
    enabled = models.BooleanField(default=True)
```

and the migration to this:
```
import uuid

from django.db import migrations
from django.db import models


def create_uuid(apps, schema_editor):
    awsenabledcategorykeys = apps.get_model("reporting", "awsenabledcategorykeys")
    for category in awsenabledcategorykeys.objects.all():
        category.uuid = uuid.uuid4()
        category.save()


class Migration(migrations.Migration):

    dependencies = [
        ("reporting", "0288_auto_20230615_1658"),
    ]

    operations = [
        migrations.AddField(
            model_name="awsenabledcategorykeys",
            name="uuid",
            field=models.UUIDField(default=uuid.uuid4, editable=False, null=True),
        ),
        migrations.RunPython(create_uuid, reverse_code=migrations.RunPython.noop),
        migrations.AlterField(
            model_name="awsenabledcategorykeys",
            name="key",
            field=models.CharField(max_length=253),
        ),
        migrations.AlterField(
            model_name="awsenabledcategorykeys",
            name="uuid",
            field=models.UUIDField(default=uuid.uuid4, editable=False, primary_key=True, serialize=False, unique=True),
        ),
    ]

```

you end up with this:
```
postgres=# \d+ reporting_awsenabledcategorykeys;
                            Table "org1234567.reporting_awsenabledcategorykeys"
 Column  |          Type          | Collation | Nullable | Default | Storage  | Stats target | Description 
---------+------------------------+-----------+----------+---------+----------+--------------+-------------
 key     | character varying(253) |           | not null |         | extended |              | 
 enabled | boolean                |           | not null |         | plain    |              | 
 uuid    | uuid                   |           | not null |         | plain    |              | 
Indexes:
    "reporting_awsenabledcategorykeys_uuid_0cd0c5ec_pk" PRIMARY KEY, btree (uuid)
    "reporting_awsenabledcategorykeys_key_enabled_idx" btree (key, enabled)
    "reporting_awsenabledcategorykeys_key_idx" btree (key varchar_pattern_ops)
    "reporting_awsenabledcategorykeys_uuid_0cd0c5ec_uniq" UNIQUE CONSTRAINT, btree (uuid)
Access method: heap
```
which is what we are after. With this migration, we create the new UUID field, populate it, and then swap the keys in place.

Reply:
The key will still need to be unique but that was an easy enough change to add in, I was trying to create the uuid with:
```
migrations.RunSQL("UPDATE reporting_awsenabledcategorykeys SET uuid = uuid_generate_v4();"),
```
But kept running into duplicate kye errors, changing that to python seemed to be the trick. 🙇